fairlearn/datasets/_fetch_acspubliccoverage.py

```diff
@@ -0,0 +1,146 @@
+                            ):
+    """Load the ACS Public Coverage dataset.
+
+    Download it if necessary.
+
+    ==============   ==============
+    Samples total         1,138,289
+    Dimensionality               19
+    Features                   real
+    Target                     real
+    ==============   ==============
+
+    Source: Paper: Frances Ding et al. (2021) [1]_
+            and corresponding repository https://github.com/zykls/folktables/
+
+    .. versionadded:: 0.7.1
+
+    Prediction task is to determine whether a person has public
+    health coverage.
+
+    Parameters
+    ----------
+    cache : bool, default=True
+        Whether to cache downloaded datasets using joblib.
+
+    data_home : str, default=None
+        Specify another download and cache folder for the datasets.
+        By default, all fairlearn data is stored in '~/.fairlearn-data'
+        subfolders.
+
+    as_frame : bool, default=False
+        If True, the data is a pandas DataFrame including columns with
+        appropriate dtypes (numeric, string or categorical). The target is
+        a pandas DataFrame or Series depending on the number of target_columns.
+        The Bunch will contain a ``frame`` attribute with the target and the
+        data. If ``return_X_y`` is True, then ``(data, target)`` will be pandas
+        DataFrames or Series as describe above.
+
+    return_X_y : bool, default=False
+        If True, returns ``(data.data, data.target)`` instead of a Bunch
+        object.
+
+    states : list, default=None
+        Filters for data points only from these states.
+        If None, returns data from all 50 US states, including Puerto Rico.
+
+    Returns
+    -------
+    dataset : :obj:`~sklearn.utils.Bunch`
+        Dictionary-like object, with the following attributes.
+
+        data : ndarray, shape (1138289, 20)
+            Each row corresponding to the 19 feature values in order as well as
+            the target variable (PUBCOV).
+            If ``as_frame`` is True, ``data`` is a pandas object.
+        target : numpy array of shape (1138289,)
+            Each value represents whether the person has public health coverage.
+            If ``as_frame`` is True, ``target`` is a pandas object.
+        feature_names : list of length 20
+            Array of ordered feature names used in the dataset.
+        DESCR : string
+            Description of the ACS Public Coverage dataset.
+
+    (data, target) : tuple of (numpy.ndarray, numpy.ndarray)
+        if ``return_X_y`` is True and ``as_frame`` is False
+
+    (data, target) : tuple of (pandas.DataFrame, pandas.Series)
+        if ``return_X_y`` is True and ``as_frame`` is True
+
+    References
+    ----------
+    .. [1] Ding, F., Hardt, M., Miller, J., & Schmidt, L. (2021).
+       "Retiring Adult: New Datasets for Fair Machine Learning."
+       Advances in Neural Information Processing Systems, 34.
+
+    """
+
+    if not data_home:
```

Comment:
There is some code to validate abbreviations in fetch above for ACSIncome.  Perhaps that can be extracted into a common utility and re-used for both datasets, once the initial PR is merged, as that validation logic is needed in this new fetch method as well.